server/service/src/sync/translations/master_list.rs

```diff
@@ -47,6 +49,35 @@ impl SyncTranslation for MasterListTranslation {
         };
         Ok(PullTranslateResult::upsert(result))
     }
+
+    // Soft deletes were implemented in OG months after program requisitions was
+    // rolled out, so previously hard deleted records may be gone even if they
+    // are linked to program. Recreate these records to keep the FK link.
+    fn try_translate_from_delete_sync_record(
+        &self,
+        connection: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<PullTranslateResult, anyhow::Error> {
+        let program =
+            ProgramRowRepository::new(connection).find_one_by_id(&sync_record.record_id)?;
+        let master_list =
+            MasterListRowRepository::new(connection).find_one_by_id(&sync_record.record_id)?;
+
+        if let (Some(_), Some(master_list)) = (program, master_list) {
+            let result = MasterListRow {
+                id: master_list.id.clone(),
+                name: master_list.name.clone(),
+                code: master_list.code.clone(),
+                description: master_list.description.clone(),
+                is_active: false,
+            };
+            return Ok(PullTranslateResult::upsert(result));
+        }
+
+        Ok(PullTranslateResult::Ignored(
```

Comment:
We did have delete for master list before... but can't remember if it was before or after soft deletes were implemented in mSupply.... should I account for this? 🤔

Reply:
Why do you look for the program row here? Shouldn't you always soft delete the master_list row even if there isn't a program associate with it?

Reply:
master_list row isn't a foreign key on anything just the programs (that's also a master list but a separate one) ...if I recall correctly?

Reply:
I guess it can only be deleted in mSupply when its not used anywhere?

Or only softdelete the masterlist if a program row uses it otherwise do a hard delete? Think, this would also make sense to me when reading the code without having too much background.

Btw. the comment "Recreate these records to keep the FK link." is also misleading since you are not recreating anything...

Reply:
Before soft deletes was implemented a few months after OMS programs, it was all hard delete. mSupply doesn't really handle foreign keys, so it isn't a problem there. 

Yeah... Can't recreate since the delete record only has ID...